FirebaseAdmin/FirebaseAdmin.IntegrationTests/Auth/TenantAwareFirebaseAuthTest.cs

```diff
@@ -0,0 +1,379 @@
+            var updateArgs = new UserRecordArgs
+            {
+                Uid = uid,
+                DisplayName = "Updated Name",
+                Email = randomUser.Email,
+                PhoneNumber = randomUser.PhoneNumber,
+                PhotoUrl = "https://example.com/photo.png",
+                EmailVerified = true,
+                Password = "secret",
+            };
+            user = await this.auth.UpdateUserAsync(updateArgs);
+            Assert.Equal(uid, user.Uid);
+            Assert.Equal(randomUser.Email, user.Email);
+            Assert.Equal(randomUser.PhoneNumber, user.PhoneNumber);
+            Assert.Equal("Updated Name", user.DisplayName);
+            Assert.Equal("https://example.com/photo.png", user.PhotoUrl);
+            Assert.True(user.EmailVerified);
+            Assert.False(user.Disabled);
+            Assert.NotNull(user.UserMetaData.CreationTimestamp);
+            Assert.Null(user.UserMetaData.LastSignInTimestamp);
+            Assert.Equal(2, user.ProviderData.Length);
+            Assert.Empty(user.CustomClaims);
+            Assert.Equal(this.tenantId, user.TenantId);
+
+            // Get user by email
+            user = await this.auth.GetUserByEmailAsync(randomUser.Email);
+            Assert.Equal(uid, user.Uid);
+            Assert.Equal(this.tenantId, user.TenantId);
+
+            // Disable user and remove properties
+            var disableArgs = new UserRecordArgs
+            {
+                Uid = uid,
+                Disabled = true,
+                DisplayName = null,
+                PhoneNumber = null,
+                PhotoUrl = null,
+            };
+            user = await this.auth.UpdateUserAsync(disableArgs);
+            Assert.Equal(uid, user.Uid);
+            Assert.Equal(randomUser.Email, user.Email);
+            Assert.Null(user.PhoneNumber);
+            Assert.Null(user.DisplayName);
+            Assert.Null(user.PhotoUrl);
+            Assert.True(user.EmailVerified);
+            Assert.True(user.Disabled);
+            Assert.NotNull(user.UserMetaData.CreationTimestamp);
+            Assert.Null(user.UserMetaData.LastSignInTimestamp);
+            Assert.Single(user.ProviderData);
+            Assert.Empty(user.CustomClaims);
+            Assert.Equal(this.tenantId, user.TenantId);
+
+            // Delete user
+            await this.auth.DeleteUserAsync(uid);
+            var exception = await Assert.ThrowsAsync<FirebaseAuthException>(
+                async () => await this.auth.GetUserAsync(uid));
+
+            Assert.Equal(AuthErrorCode.UserNotFound, exception.AuthErrorCode);
+        }
+
+        [Fact]
+        public async Task DeleteUsers()
+        {
+            var user1 = await this.userBuilder.CreateRandomUserAsync();
+            var user2 = await this.userBuilder.CreateRandomUserAsync();
+            var user3 = await this.userBuilder.CreateRandomUserAsync();
+
+            await Task.Delay(millisecondsDelay: 1000);
```

Comment:
I've just copy-pasted this bit from the existing test case in `FirebaseAuthTest`. In the next PR I will de-duplicate some of this. For now I'm going to keep the named params as is, since it's not causing any issues.